x-pack/plugins/stack_connectors/server/connector_types/inference/inference.ts

```diff
@@ -181,16 +182,26 @@ export class InferenceConnector extends SubActionConnector<Config, Secrets> {
    * @signal abort signal
    */
   public async performApiUnifiedCompletionStream(params: UnifiedChatCompleteParams) {
-    return await this.esClient.transport.request<UnifiedChatCompleteResponse>(
+    const response = await this.esClient.transport.request<UnifiedChatCompleteResponse>(
       {
         method: 'POST',
         path: `_inference/completion/${this.inferenceId}/_unified`,
         body: { ...params.body, n: undefined }, // exclude n param for now, constant is used on the inference API side
       },
       {
         asStream: true,
+        meta: true,
+        signal: params.signal,
       }
     );
+
+    // errors should be thrown as it will not be a stream response
+    if (response.statusCode >= 400) {
+      const error = await streamToString(response.body as unknown as Readable);
+      throw new Error(error);
+    }
```

Comment:
Errors from the API call were not caught and simply streamed to the consumer, which was fairly problematic.

Fixed by checking the status code and throwing for error code.

The response format remain unchanged for successful calls (only returning the streaming body)